oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/VersionGarbageCollector.java

```diff
@@ -1355,6 +1356,7 @@ private void collectUnmergedBranchCommits(final NodeDocument doc, final GCPhases
             
             olderUnmergedBranchCommits.forEach(bcRevision -> removeUnmergedBCRevision(bcRevision, doc, updateOp));
             deletedUnmergedBCSet.addAll(olderUnmergedBranchCommits);
+            fullGCStats.collectedUnmergedBranchCommits(GCPhase.FULL_GC_COLLECT_UNMERGED_BC, olderUnmergedBranchCommits.size());
```

Comment:
same as above, collecting deletion stats here is wrong.

Reply:
They may seem duplicated but are not, the new methods count the amounts of garbage that would be deleted by each of the phases, and the idea is to eventually use this as a reference to know which phases of FullGC would bring more benefit.

Reply:
Probably all the methods could be renamed to state this better.